### PR TITLE
*: use std/slices to replace exp/slices

### PR DESCRIPTION
**Pull request by hawkingrei**
### What problem does this PR solve?

Issue Number: ref #45933

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/restore/client.go [hunk 1]
```diff
@@ -2822,26 +2823,17 @@ func (rc *Client) InitSchemasReplaceForDDL(
...
-			return false
+		if c := cmp.Compare(i.GetMaxTs(), j.GetMaxTs()); c != 0 {
+			return c
 		}
-
-		if i.GetResolvedTs() < j.GetResolvedTs() {
-			return true
-		} else if i.GetResolvedTs() > j.GetResolvedTs() {
-			return false
+		if c := cmp.Compare(i.GetResolvedTs(), j.GetResolvedTs()); c != 0 {
```
Comment (Defined2014) [inline br/pkg/restore/client.go:2833]: Line 2833-2836 could direclty return `cmp.Compare(i.GetResolvedTs(), j.GetResolvedTs())`

**Review by Defined2014 — APPROVED** [review]
LGTM, pls fix compile

File: br/pkg/restore/client.go (hunk 1 above)
Comment (hawkingrei) [reply br/pkg/restore/client.go]: Done

**Comment by hawkingrei** [comment]
> LGTM, pls fix compile

Fixed

**Review by hi-rustin — APPROVED** [review]
:shipit:

**Comment by StevenACoffman** [comment]
This closes #45629

**Pull request merged**